src/main/java/org/broadinstitute/hellbender/utils/read/ReadCoordinateComparator.java

```diff
@@ -54,10 +60,9 @@ private int compareCoordinates( final GATKRead first, final GATKRead second ) {
         final int firstRefIndex = ReadUtils.getReferenceIndex(first, header);
         final int secondRefIndex = ReadUtils.getReferenceIndex(second, header);
 
-        if ( first.isUnmapped() ) {
-            return second.isUnmapped() ? 0 : 1;
-        }
-        else if ( second.isUnmapped() ) {
+        if ( firstRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
+            return secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ? 0 : 1;
+        } else if ( secondRefIndex == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX ) {
```

Comment:
but at this point isUnmapped is already called. so calling it again seems bogus
also, the behavior is different: in the case of 'mapped' read with -1 as getReferenceIndex, proposed code mimics SAMRecordCoordinateComparator while the existing code does not.